CAIPs/caip-74.md

```diff
@@ -1,144 +1,271 @@
+```verilog
+
+type Prinicpal Bytes // a multidid
+type Varsig Bytes
+type Resource String // URL
+type Ability String // e.g. crud/create
+
+type NB { String : Any }
+type Abilities { Ability : [NB] }
+type Resources { Resource : Abilities }
+type Fact { String: Any }
 
-```
 type CACAO struct {
-  h Header // container meta-information
-  p Payload // payload
-  s Signature // signature, single
+  iss Principal
+  aud Principal
+  s Varsig
+  
+  v String
+  att Resources
+  nnc String
+  prf optional [&CACAO]
+  iat optional Int
+  nbf optional Int
+  exp optional Int
+  fct optional Fact
 }
+
 ```
 
-Header uniquely identifies the payload format:
+### Decode to IPLD
 
-```
-type Header struct {
-  t String // specifies format of the payload
-}
-```
+This section describes how to convert different object-capability formats into the IPLD schema described above.
 
-The header type will be `caip122` in reference to the [CAIP-122]() specification for the SIWx data model. In an [older version of the specification](https://github.com/ChainAgnostic/CAIPs/blob/91aaaff73038c2629ff11b88c2209f61521d8ece/CAIPs/caip-74.md), the header type was restricted to `eip4361` as it was designed to work only with Sign-in with Ethereum. As such, newer implementations MUST be able to deal with both header types appropriately.
+#### SIWx + ReCap
 
-The payload structure must be presented as follows:
+The following values can be easily translated from the SIWx (CAIP-122) specification:
 
-```
-type Payload struct {
-  domain String // =domain
-  iss String // = DID pkh
-  aud String // =uri
-  version String
-  nonce String
-  iat String // RFC3339 date-time =issued-at
-  nbf optional String // RFC3339 date-time =not-before
-  exp optional String // RFC3339 date-time = expiration-time
-  statement optional String // =statement
-  requestId optional String // =request-id
-  resources optional [ String ] // =resources as URIs
-}
-```
+* `iss` - a multidid encoded DID PKH constructued using `address` and `chain-id`
+* `aud` - a multidid encoded DID based on `uri`
+* `v` - set to `version`
+* `nnc` - set to `nonce`
 
-It is important to note, that issuer here is [did:pkh](https://github.com/w3c-ccg/did-pkh/blob/main/did-pkh-method-draft.md), which includes both blockchain address and blockchain network information.
-Also, as per [CAIP-122]() specificaction,`iat`, `nbf`, and `exp` are encoded as [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339#section-5.6) `date-time`, which could include milliseconds precision.
+**Timestamps**
 
-The signature in essence is just bytes, but we have to give a hint on how the signature verification should work. The signature verification type is referenced from methods that are listed as possible within the [CAIP-122]() namespace.
+* `iat` - is based on `issued-at`
+* `nbf` - is based on `not-before`
+* `exp` - is based on `expiration-time`
+* `fct.z-iat` - timezone info from `issued-at`
```

Comment:
Timezones mandatory? Default to 0/UTC if not present for backwards compat simplicity?

Reply:
I think if there is no timezone in the SIWx and no ms then they can be left out. This will be specified in Appendix A.